PHPCompatibility/Sniffs/Syntax/NewConstantDereferencingSniff.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2021 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Sniffs\Syntax;
+
+use PHPCompatibility\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * Detect constant dereferencing.
+ *
+ * As of PHP 5.6, constants can now be dereferenced to access individual elements and characters.
+ *
+ * As of PHP 8.0, constants can also be dereferenced to invoke method calls, though this support only applies to syntax and must be used in conjunction with the scalar objects extension to work at runtime.
```

Comment:
Let's try and keep the lines semi-readable.
```suggestion
 * As of PHP 8.0, constants can also be dereferenced to invoke method calls, though this support
 * only applies to syntax and must be used in conjunction with the scalar objects extension to work at runtime.
```